### PR TITLE
checking extension before reading to check outputs

### PR DESCRIPTION
**Pull request by AakashGfude**
Checking if the extension is `ipynb` before assuming it is a notebook and reading the cells for outputs.

### CODE REVIEW

**Comment by AakashGfude** [comment]
Hey @chrisjsewell , do you know how to counter the following error causing CI build fail?
image

**Comment by AakashGfude** [comment]
closes https://github.com/ExecutableBookProject/MyST-NB/issues/130 .
Regarding files with different extensions but same name in toc, we can create a new issue.

File: myst_nb/cache.py [hunk 1]
```diff
@@ -212,7 +211,12 @@ def execute_staged_nb(cache_base, pk_list):
 
 def _read_nb_output_cells(source_path, jupyter_execute_notebooks):
     has_outputs = False
-    if jupyter_execute_notebooks and jupyter_execute_notebooks == "auto":
+    ext = source_path[source_path.rfind(".") :]
```
Comment (chrisjsewell) [inline myst_nb/cache.py:214]: Probably to use `os.path.splitext` here

**Comment by chrisjsewell** [comment]
> Hey @chrisjsewell , do you know how to counter the following error causing CI build fail?

Yeh weird, the token is supplied https://github.com/ExecutableBookProject/MyST-NB/blob/b64cc307b38c14db469e410434895f13f63773ff/.github/workflows/tests.yml#L51

But perhaps this only works if the PR is from a branch on this repo??

**Comment by chrisjsewell** [comment]
Ah I see there is an actual GitHub coveralls action: https://github.com/marketplace/actions/coveralls-github-action, that will hopefully fix this

**Comment by chrisjsewell** [comment]
I'll try this out in a separate PR in a min

File: myst_nb/cache.py (hunk 1 above)
Comment (AakashGfude) [reply myst_nb/cache.py]: Done

**Comment by mmcky** [comment]
thanks @AakashGfude and @chrisjsewell

**Comment by mmcky** [comment]
I will try out the latest `master` on `myst-nb.example-project`

**Pull request merged**